studio/schemas/documents/compensations.ts

```diff
@@ -15,24 +16,29 @@ const compensations = defineType({
   type: "document",
   title: "Compensations",
   fields: [
-    locations,
     title,
     titleSlug,
-    seo,
+    locations,
     defineField({
-      name: "showSalaryCalculator",
-      title: "Show Salary Calculator",
-      description: "Should the salary calculator be visible on the page?",
-      type: "boolean",
-      initialValue: true,
+      ...compensationDetails,
+      readOnly: ({ parent }) => !parent?.showEstimatedSalary,
     }),
+    {
+      name: "showEstimatedSalary",
+      title: "Show Estimated Salary",
+      type: "boolean",
+      description:
+        "Toggle this option to show or hide the estimated salary for the locations you've selected. Disable this if the feature isn't ready or if you prefer not to display salary details at this time.",
+      initialValue: false,
+    },
     defineField({
```

Comment:
Why is benefits not placed within compensationDetails?

Reply:
Because benefits are text components not related to data like salary and pension

Reply:
It says "financial benefits" in the description though 🤔 I think the main issue is that "Compensation Details" is very generic. If the whole section is disabled by "Show estimated salary", maybe we could call it "Salary Details" instead?

Reply:
"Benefits" and "Financial benefits" are similar but have different definitions still. One is broader and has wider range of perks and advantages provided by an employer, which may include both financial and non-financial elements. While the other refers specifically to benefits that have a direct monetary value like salary, bonus, pension. "Salary details" would refer to the base salary information, and wouldn't really cover pension or bonus.

Reply:
If "salary" does not cover pension or bonus (which makes sense), then why does disabling "Show estimated salary" make the pension and bonus fields read-only?

Reply:
I added "estimated Salary" more as the title for the feature.